### DISCUSSION TITLE
How can I add extra data (First name, Last name, role) to AuthUser

### DISCUSSION THREAD

**Discussion by vpaflah** [gladly-team/next-firebase-auth#314] [Q&A]
I would like to add some additional data to the AuthUser from firestore doc, How can I do that.

**Comment by kmjennison** [comment]
You'll want to store that info in a database, which is functionality not related to this library. You could choose to use Firebase claims, but it's probably not a great idea to attach data there. From [the docs](https://firebase.google.com/docs/auth/admin/custom-claims#best_practices_for_custom_claims):

>Custom claims are only used to provide access control. They are not designed to store additional data (such as profile and other custom data). While this may seem like a convenient mechanism to do so, it is strongly discouraged as these claims are stored in the ID token and could cause performance issues because all authenticated requests always contain a Firebase ID token corresponding to the signed in user.

**Reply by vpaflah** [reply]
I checked this solution, but it's not a correct method to follow.
Is there any other way to achive the solution.

And thanks for your response <3

**Reply by agawrylak** [reply]
How could I achieve a form where I login using next-firebase-auth and also store additional properties in my DB?
